stan/math/gpu/matrix_gpu.hpp

```diff
@@ -158,6 +162,99 @@ class matrix_gpu {
     oclBuffer_ = a.buffer();
     return *this;
   }
+
+  /**
+   * Stores zeros in the matrix on the GPU.
+   * Supports writing zeroes to the lower and upper triangular or
+   * the whole matrix.
+   *
+   * @tparam triangular_view Specifies if zeros are assigned to
+   * the entire matrix, lower triangular or upper triangular with
+   * possible values stan::math:Entire, stan::math:Lower or
+   * stan::math::Upper
+   *
+   */
+  template <int triangular_view>
```

Comment:
Question:

Would it make sense to have a default for the template parameter? Perhaps `stan::math::Entire`?

Reply:
I think that makes perfect sense. I added gpu::Entire here.